node/src/components/block_synchronizer/execution_results_acquisition.rs

```diff
@@ -315,18 +323,24 @@ impl ExecutionResultsAcquisition {
             ) => {
                 debug!("apply_block_execution_results_or_chunk: (Pending, ChunkWithProof)");
                 match apply_chunk(block_hash, checksum, HashMap::new(), chunk, None) {
+                    Ok(ApplyChunkOutcome::HadIt { .. }) => {
+                        error!("cannot have already had chunk if in pending mode");
+                        return Err(Error::InvalidOutcomeFromApplyingChunk { block_hash });
```

Comment:
I'm wondering if this is an error. I didn't dig into it much, but it seems like it's legit that such a situation may happen in the legit flow and the node will continue working, so we shouldn't avoid having `ERROR` for it in the log.

Reply:
I actually think it's an error. If a checksum is received, the acquisition switches to `Pending` and it waits for the first chunk/results to be applied. If it's in `Pending` should mean that it shouldn't already have chunks/results, so if we apply a chunk and it was already applied means that we have a programmer error of some sorts.

Reply:
Yes, as @alsrdn said, this is a true programmer error.

Reply:
Yup, agreed. I forgot we have a separate `Acquiring` state for the actual acquisition process.